src/LayoutButton/LayoutButton.tsx

```diff
@@ -1,16 +1,30 @@
 import type { ButtonBaseProps } from "@material-ui/core/ButtonBase";
+
 import ButtonBase from "@material-ui/core/ButtonBase";
 import clsx from "clsx";
 import React from "react";
 
 import useStyles from "./styles";
 
-export const LayoutButton: React.FC<ButtonBaseProps> = React.forwardRef(
-  ({ className, children, ...rest }, ref) => {
+export type LayoutButtonProps<T extends React.ElementType = "button"> =
+  ButtonBaseProps<T> & {
+    state?: "default" | "hover" | "active";
```

Comment:
I would extract `"default" | "hover" | "active"` to a separate type. Also, I would add `"disabled"` state